lib/syskit/task_context.rb

```diff
@@ -672,6 +672,8 @@ def read_current_state
         # Returns true if this component needs to be setup by calling the
         # #setup method, or if it can be used as-is
         def ready_for_setup?(state = nil)
+            return running? if read_only?
```

Comment:
This is wrong. `running?` refers to the Roby task. When we check if we should set it up, `running?` will always be false.

Look later in this very method. There is a check for the component state to be an exception state, STOPPED or PRE_OPERATIONAL. You would need to check whether the state is a running state in the case of read-only

And add tests ...